src/app/qgsdiagramproperties.cpp

```diff
@@ -193,6 +196,9 @@ QgsDiagramProperties::QgsDiagramProperties( QgsVectorLayer* layer, QWidget* pare
     mDiagramFrame->setEnabled( false );
     mFixedSizeRadio->setChecked( true );
     mDiagramUnitComboBox->setCurrentIndex( mDiagramUnitComboBox->findText( tr( "mm" ) ) );
+    mDiagramLineUnitComboBox->blockSignals( true );
+    mDiagramLineUnitComboBox->setUnit( QgsSymbolV2::MM );
```

Comment:
Actually I don't think this is needed. Nothing is connected to that widget, so you can remove the blocks.